framework/sac_rlhf.py

```diff
@@ -489,6 +501,15 @@ def get_action(self, x):
                 writer.add_scalar("charts/SPS", int(global_step / (time.time() - start_time)), global_step)
                 if args.autotune:
                     writer.add_scalar("losses/alpha_loss", alpha_loss.item(), global_step)
+                if args.measure_performance == "pearson":
+                    metrics.add_rewards(
+                        np.concatenate(reward_net(data.observations, data.actions).detach().cpu().numpy(), axis=0),
+                        np.concatenate(data.rewards.detach().cpu().numpy(), axis=0)
+                    )
+                    pearson_corr = metrics.compute_pearson_correlation()
+                    writer.add_scalar("charts/pearson_correlation", pearson_corr, global_step)
+                    print(f"Pearson Correlation: {pearson_corr}")
+                    # metrics.reset()
```

Comment:
I would prefer to not comment this, this could lead to weird behaviour in the unlikely case one wants to try it this metric somewhere else in the code. Even if this does not happen, I see no downside of just doing this always.
But I see the point, so do it as you like.

We should keep in mind that the meaning of reward changes by merging #18